### DISCUSSION TITLE
Projection/Unprojection of points not considering map repeat mode across 180th Meridian

### DISCUSSION THREAD

**Discussion by Anand2489** [visgl/deck.gl#6972] [Q&A]
### Description

Projection/Unprojection of points not considering map repeat mode with `wrapLongitude` across 180th Meridian. When trying to render TextLayer parallel to a line(crossing Pacific ocean) at an orthogonal offset, it's not rendering the co-ordinates on the correct side of the map

Any configuration to enable it ?

### Steps to Reproduce

<img width="1371" alt="Screenshot 2022-06-02 at 5 53 08 PM" src="https://user-images.githubusercontent.com/5299907/171629286-3f2bdacc-d4cd-49cd-99dd-1e3f1ffb2098.png">

### Code pen

https://codepen.io/andycool/pen/ExQbbeG?editors=001

**Comment by Pessimistress** [comment]
The TextLayer can't possibly know that it is attached to a "shortest path" of some sort. You need to return the correct coordinate with `getPosition`.

**Reply by Anand2489** [reply]
@Pessimistress thanks for the reply :) 

I understand that we will have to provide correct coordinates via `getPosition` method to text layer but how can we achieve that. I tried using projection & unProjection method but it didn't work. Following is a simplified version of the calculation

```js
const srcXY = this.context.viewport.project(srcLngLat)
const desXY = this.context.viewport.project(destLngLat)
const labelXY = [(srcXY[0] + desXY[0]) / 2, (srcXY[1] + desXY[1]) / 2]

const labelCoordinate = this.context.viewport.unproject(labelXY)

```

Could you suggest some alternate way which works generically for lines crossing 180th AntiMeridian or not. I could update the codepen too in case you want a demo!

**Reply by Pessimistress** [reply]
```js
const deltaLng = Math.abs(srcLngLat[0] - destLngLat[0]);
const labelLng = (srcLngLat[0] + destLngLat[0]) / 2;
if (deltaLng > 180) {
  labelLng = labelLng < 0 ? labelLng + 180 : labelLng - 180;
}
```
